### PR TITLE
Adding CONTAINERS WIP doc

### PR DESCRIPTION
**Pull request by rvs**
First cut at documenting our container efforts

### CODE REVIEW

File: docs/CONTAINERS.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Direct support of OCI containers on EVE
+
+THIS IS A PROTOTYPE -- USE (OR BETTER YET DON'T) AT YOUR OWN RISK.
+
+Now that the disclaimer is out of the way, what we're trying to build here is the direct support for [OCI containers](https://www.opencontainers.org/) on EVE. This means that you will be able to point at a random container on Docker HUB and run it on EVE as easily as you would be running a VM.
+
+With projects like [Weave Ignite](https://github.com/weaveworks/ignite) there has been quite a resurgance of interest in running containers in thin VMs lately. We're very much believers in that trend, but at the same time, since we're currently based on Xen we're using excellent work of Stefano Stabellini
```
Comment (deitch) [inline docs/CONTAINERS.md]: resurgence

File: docs/CONTAINERS.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+### The capabilities in EVE
+
+For now, we have added two of the binary artifacts to EVE:
+   * /usr/sbin/rkt
+   * /usr/sbin/stage1-xen.aci
+
+### How is it supported?
+
+It isn't. It isn't any more supported than /usr/bin/mail that happens to be part of pillar ;-) But you can star playing with it.
```
Comment (deitch) [inline docs/CONTAINERS.md]: start

File: docs/CONTAINERS.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+```
+ln -s /persist/rkt /var/lib/rkt
+rkt --insecure-options=image fetch docker://alpine
+rkt run sha512-<ID FROM PREVIOUS LINE> --interactive --insecure-options=image --stage1-path=/usr/sbin/stage1-xen.aci
+```
+
+### Why do I have to have that first line anyway?
+
+rkt wants to keep its state in /var/lib/rkt but on EVE you want it to be persistent -- hence keep it in /persist/rkt
```
Comment (deitch) [inline docs/CONTAINERS.md]: You can do `rkt --dir /persist/rkt <command> ...` to change it. It is a good idea to have the link to make it easier, but we should point it out.

Or maybe we should make the symlink part of the base image?

Also, does the dir `/persist/rkt` need to exist before running?

**Comment by rvs** [comment]
Hi @deitch -- thanks for spotting the typos/etc. I fixed those. As for your comment:
```I am concerned about using rkt, as it appears to be be rapidly losing any market share. runc works, but it doesn't have any "stage1-path" option, since it is built entirely differently.```
No disagreement there. We started with rkt because it is very selfcontained, small, easy to package and, most importantly, already had stage1 Xen support.

However, interestingly enough, @sstabellini mentioned that he may look into runc support as well. At any rate -- replacing rkt with runc should be trivial if/when we decide to do it.

File: docs/CONTAINERS.md (hunk 3 above)
Comment (rvs) [reply docs/CONTAINERS.md]: No it doesn't have to exist.

File: docs/CONTAINERS.md (hunk 2 above)
Comment (rvs) [reply docs/CONTAINERS.md]: fixed

**Comment by sstabellini** [comment]
Yes, porting stage1-xen to the runc interface is pretty easy, and it is in the plans. In fact, I even have a PoC already. Stay tuned ;-)

**Pull request merged**